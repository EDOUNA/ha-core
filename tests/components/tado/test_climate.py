"""The sensor tests for the Tado platform."""

from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant

from .util import async_init_integration


async def test_air_con(hass: HomeAssistant, snapshot: SnapshotAssertion) -> None:
    """Test creation of aircon climate."""

    await async_init_integration(hass)

    state = hass.states.get("climate.air_conditioning")
    assert state.state == "cool"

    # Only test for a subset of attributes in case
    # HA changes the implementation and a new one appears
    expected_keys = [
        "current_humidity",
        "current_temperature",
        "fan_mode",
        "fan_modes",
        "friendly_name",
        "hvac_action",
        "hvac_modes",
        "max_temp",
        "min_temp",
        "preset_mode",
        "preset_modes",
        "supported_features",
        "target_temp_step",
        "temperature",
    ]

    actual_attributes_subset = {key: state.attributes.get(key) for key in expected_keys}
    assert actual_attributes_subset == snapshot


async def test_heater(hass: HomeAssistant, snapshot: SnapshotAssertion) -> None:
    """Test creation of heater climate."""

    await async_init_integration(hass)

    state = hass.states.get("climate.baseboard_heater")
    assert state.state == "heat"

    # Only test for a subset of attributes in case
    # HA changes the implementation and a new one appears
    expected_keys = [
        "current_humidity",
        "current_temperature",
        "friendly_name",
        "hvac_action",
        "hvac_modes",
        "max_temp",
        "min_temp",
        "preset_mode",
        "preset_modes",
        "supported_features",
        "target_temp_step",
        "temperature",
    ]

    actual_attributes_subset = {key: state.attributes.get(key) for key in expected_keys}
    assert actual_attributes_subset == snapshot


async def test_smartac_with_swing(
    hass: HomeAssistant, snapshot: SnapshotAssertion
) -> None:
    """Test creation of smart ac with swing climate."""

    await async_init_integration(hass)

    state = hass.states.get("climate.air_conditioning_with_swing")
    assert state.state == "auto"

    # Only test for a subset of attributes in case
    # HA changes the implementation and a new one appears
    expected_keys = [
        "current_humidity",
        "current_temperature",
        "fan_mode",
        "fan_modes",
        "friendly_name",
        "hvac_action",
        "hvac_modes",
        "max_temp",
        "min_temp",
        "preset_mode",
        "preset_modes",
        "supported_features",
        "target_temp_step",
        "temperature",
        "swing_modes",
    ]

    actual_attributes_subset = {key: state.attributes.get(key) for key in expected_keys}
    assert actual_attributes_subset == snapshot


async def test_smartac_with_fanlevel_vertical_and_horizontal_swing(
    hass: HomeAssistant, snapshot: SnapshotAssertion
) -> None:
    """Test creation of smart ac with swing climate."""

    await async_init_integration(hass)

    state = hass.states.get("climate.air_conditioning_with_fanlevel")
    assert state.state == "heat"

    # Only test for a subset of attributes in case
    # HA changes the implementation and a new one appears
    expected_keys = [
        "current_humidity",
        "current_temperature",
        "fan_mode",
        "fan_modes",
        "friendly_name",
        "hvac_action",
        "hvac_modes",
        "max_temp",
        "min_temp",
        "preset_mode",
        "preset_modes",
        "supported_features",
        "target_temp_step",
        "temperature",
        "swing_modes",
    ]

    actual_attributes_subset = {key: state.attributes.get(key) for key in expected_keys}
    assert actual_attributes_subset == snapshot
