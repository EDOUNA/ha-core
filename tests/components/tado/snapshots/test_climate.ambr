# serializer version: 1
# name: test_air_con
  dict({
    'current_humidity': 60.9,
    'current_temperature': 24.8,
    'fan_mode': 'auto',
    'fan_modes': list([
      'auto',
      'high',
      'medium',
      'low',
    ]),
    'friendly_name': 'Air Conditioning',
    'hvac_action': <HVACAction.COOLING: 'cooling'>,
    'hvac_modes': list([
      <HVACMode.OFF: 'off'>,
      <HVACMode.AUTO: 'auto'>,
      <HVACMode.HEAT: 'heat'>,
      <HVACMode.COOL: 'cool'>,
      <HVACMode.HEAT_COOL: 'heat_cool'>,
      <HVACMode.DRY: 'dry'>,
      <HVACMode.FAN_ONLY: 'fan_only'>,
    ]),
    'max_temp': 31.0,
    'min_temp': 16.0,
    'preset_mode': 'auto',
    'preset_modes': list([
      'away',
      'home',
      'auto',
    ]),
    'supported_features': <ClimateEntityFeature: 409>,
    'target_temp_step': 1,
    'temperature': 17.8,
  })
# ---
# name: test_heater
  dict({
    'current_humidity': 45.2,
    'current_temperature': 20.6,
    'friendly_name': 'Baseboard Heater',
    'hvac_action': <HVACAction.IDLE: 'idle'>,
    'hvac_modes': list([
      <HVACMode.OFF: 'off'>,
      <HVACMode.AUTO: 'auto'>,
      <HVACMode.HEAT: 'heat'>,
    ]),
    'max_temp': 31.0,
    'min_temp': 16.0,
    'preset_mode': 'auto',
    'preset_modes': list([
      'away',
      'home',
      'auto',
    ]),
    'supported_features': <ClimateEntityFeature: 401>,
    'target_temp_step': 1,
    'temperature': 20.5,
  })
# ---
# name: test_smartac_with_fanlevel_vertical_and_horizontal_swing
  dict({
    'current_humidity': 70.9,
    'current_temperature': 24.3,
    'fan_mode': 'high',
    'fan_modes': list([
      'high',
      'medium',
      'auto',
      'low',
    ]),
    'friendly_name': 'Air Conditioning with fanlevel',
    'hvac_action': <HVACAction.HEATING: 'heating'>,
    'hvac_modes': list([
      <HVACMode.OFF: 'off'>,
      <HVACMode.AUTO: 'auto'>,
      <HVACMode.HEAT: 'heat'>,
      <HVACMode.COOL: 'cool'>,
      <HVACMode.HEAT_COOL: 'heat_cool'>,
      <HVACMode.DRY: 'dry'>,
      <HVACMode.FAN_ONLY: 'fan_only'>,
    ]),
    'max_temp': 31.0,
    'min_temp': 16.0,
    'preset_mode': 'auto',
    'preset_modes': list([
      'away',
      'home',
      'auto',
    ]),
    'supported_features': <ClimateEntityFeature: 441>,
    'swing_modes': list([
      'vertical',
      'horizontal',
      'both',
      'off',
    ]),
    'target_temp_step': 1.0,
    'temperature': 25.0,
  })
# ---
# name: test_smartac_with_swing
  dict({
    'current_humidity': 42.3,
    'current_temperature': 20.9,
    'fan_mode': 'auto',
    'fan_modes': list([
      'auto',
      'high',
      'medium',
      'low',
    ]),
    'friendly_name': 'Air Conditioning with swing',
    'hvac_action': <HVACAction.HEATING: 'heating'>,
    'hvac_modes': list([
      <HVACMode.OFF: 'off'>,
      <HVACMode.AUTO: 'auto'>,
      <HVACMode.HEAT: 'heat'>,
      <HVACMode.COOL: 'cool'>,
      <HVACMode.HEAT_COOL: 'heat_cool'>,
      <HVACMode.DRY: 'dry'>,
      <HVACMode.FAN_ONLY: 'fan_only'>,
    ]),
    'max_temp': 30.0,
    'min_temp': 16.0,
    'preset_mode': 'auto',
    'preset_modes': list([
      'away',
      'home',
      'auto',
    ]),
    'supported_features': <ClimateEntityFeature: 441>,
    'swing_modes': list([
      'on',
      'off',
    ]),
    'target_temp_step': 1.0,
    'temperature': 20.0,
  })
# ---
